### DISCUSSION TITLE
CSS Background-image better support with public folder assets.

### DISCUSSION THREAD

**Discussion by misbahkhalilaz** [webpack/webpack#16940] [Q&A] [answered]
## Feature request

**What is the expected behavior?**
When using the `background-image` CSS property with webpack, and the path to some asset is given like `background-image: url('any-img.png')`, webpack should first try to resolve it from `src`, but if not found, webpack should consider it as a public asset which would be served from the public folder.

**What is motivation or use case for adding/changing the behavior?**
Suppose we've configured a web app with webpack, and we've a nicely working public folder, and all public folder assets are directly served at `/` (root) of the application. But when I try to use the public assets as `background-image` in CSS (eg: `background-image: url('/assets/bg.png')`) it doesn't work, because it just tries to resolve it in `src`. I think it is clearly mentioned in the syntax (`background-image: url('/assets/bg.png')`) that path given is a URL, but still, webpack tries to load it from src, instead of considering it a URL.

**How should this be implemented in your opinion?**

Note: this might be an issue with my webpack implementation, kindly suggest me some resources if it is already possible with webpack. Thanks!

**Are you willing to work on this yourself?**
yes

**Comment by Saeeda-kh** [comment]
I am also facing this issue for 2 days. May be possible it is an issue with the my webpack configuration. I need a clear guide for it please.

**Comment by alexander-akait** [comment, marked as answer]
You can do it using https://webpack.js.org/configuration/resolve/#resolveroots

**Reply by alexander-akait** [reply]
You can setup any number of directories where webpack will resolve

**Reply by misbahkhalilaz** [reply]
Thanks @alexander-akait, I really appreciate that 🙏

**Reply by misbahkhalilaz** [reply]
- @Saeeda-kh check this out.

**Reply by Saeeda-kh** [reply]
Thank you so much **@alexander-akai**t. it worked perfect.
